mne/coreg.py

```diff
@@ -1223,3 +1227,586 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
                 F_mri_ras, 'ras', 'ras'),
             F_ras_mni, 'ras', 'mni_tal')
     _write_fs_xfm(fname_to, T_ras_mni['trans'], kind)
+
+
+class Coregistration(object):
+    """
+    Class for MRI<->MEG coregistration.
+
+    Parameters
+    ----------
+    info : instance of Info
+        The measurement info.
+    subject : str
+        Name of the subject the data are defined for.
+    subjects_dir : path-like
+        Path to MRI subjects directory.
+
+    Notes
+    -----
+    Internal computation quantities ``parameters`` are in units of (for X/Y/Z):
+    - ``parameters[:3]`` are in radians
+    - ``parameters[3:6]`` are in m
+    - ``paramteres[6:9]`` are in scale proportion
+    """
+
+    def __init__(self, info, subject, subjects_dir):
+        self._default_parameters = [0., 0., 0., 0., 0., 0., 1., 1., 1.]
+        self._parameters = list(self._default_parameters)
+        self._last_parameters = list(self._default_parameters)
+
+        self._n_scale_params = (0, 1, 3)
+        self._n_scale_param = self._n_scale_params[0]
```

Comment:
how do we expose this option of number of scale params here?

Reply:
I actually don't think we need this `n_scale_params` at all -- we can just have the monolithic `_update_params` take into account the `self.scale_factors` being all identity (no scale), all the same and non-identity (uniform), or different (3-axis scaling). The n_scale_params is really a GUI concept/switch and I don't think it's needed here.

In terms of programatically setting things, I think we probably want a property `self.scale_factors = (1., 1., 1.)` or so, and maybe a `set_scale_factors(...)`, `set_euler_angles(...)`, `set_translation(...)` that all internally call `self._update_params(...)`